zebra-chain/src/transaction.rs

```diff
@@ -136,4 +140,12 @@ impl Transaction {
         self.inputs()
             .any(|input| matches!(input, TransparentInput::Coinbase { .. }))
     }
+
+    /// Returns the hash for the current transaction
+    pub fn hash(&self) -> TransactionHash {
+        let mut hash_writer = Sha256dWriter::default();
+        self.zcash_serialize(&mut hash_writer)
+            .expect("Transactions must serialize into the hash.");
+        TransactionHash(hash_writer.finish())
+    }
```

Comment:
Why create this method vs doing an `.into()` ?

Reply:
because the `Into` takes the Transaction by value and i needed to keep the transaction while grabbing the hash.

Reply:
OK, then maybe we could either add another `From<&Transaction>` impl, or update the existing one to only borrow, since the target `Transaction` is never modified, alá

https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/block/hash.rs#L38

Reply:
+1 on adding the `From<&Transaction>`, I'll add that in. I would like to keep the `.hash()` function though, I feel like its a lot nicer looking.

```rust
let hash = transaction.hash();

// vs

let hash: TransactionHash = (&transaction).into();
let hash = TransactionHash::from(&transaction);
```

Reply:
I'm mostly leaning towards a `From/Into` impl because there is a std [`Hash`](https://doc.rust-lang.org/std/hash/trait.Hash.html#tymethod.hash) trait but its `hash()` method signature is different; there's also [`Digest`](https://docs.rs/digest/0.9.0/digest/trait.Digest.html) but our 'hash' is the result of the activity that something that impls `Digest` does, so we don't really want to use that here anyway.

Reply:
what if we called it `to_hash()`?

https://rust-lang.github.io/api-guidelines/naming.html#c-conv

Reply:
Sure! :nailed-it: